exporter/opentelemetry-exporter-otlp-proto-http/src/opentelemetry/exporter/otlp/proto/http/metric_exporter/__init__.py

```diff
@@ -459,3 +249,15 @@ def _append_metrics_path(endpoint: str) -> str:
     if endpoint.endswith("/"):
         return endpoint + DEFAULT_METRICS_EXPORT_PATH
     return endpoint + f"/{DEFAULT_METRICS_EXPORT_PATH}"
+
+
+@deprecated(
+    version="1.16.0",
+    reason="Use one of the encoders from opentelemetry-exporter-otlp-proto-common instead",
```

Comment:
If we suggest users to use the encoder in `common`, `_get_resource_data` must be a public method in the `common` package.

Reply:
Sorry its probably not clear, I mean encoders plural as in:
* `opentelemetry.exporter.otlp.proto.common.log_encoder.encode_logs`
* `opentelemetry.exporter.otlp.proto.common.metrics_encoder.encode_metrics`
* `opentelemetry.exporter.otlp.proto.common.trace_encoder.encode_spans`

Reply:
The deprecation message is for the specific api method `get_resource_data()`. We must be specific about which method to use instead of this deprecated one. `encode_logs`, `encode_metrics` and `encode_spans` are not replacements for this function if the user were to want to use it.

Reply:
Would you deprecate it at all then? If we don't it would basically leave you with some code exercised by the core library

Reply:
I'm not saying that no one uses this API method, I just do not see any reason they would use it to not either encode logs, metrics or spans. This means they would be installing either the `http` or `grpc` package, then using this, which given where it is in the `grpc` library is some very odd no?

Reply:
That's fair. I'm fine with leaving it as is then. Please change the versioning to 1.17..0 :D